app/src/main/java/com/asfoundation/wallet/repository/ImpressionRepository.kt

```diff
@@ -0,0 +1,17 @@
+package com.asfoundation.wallet.repository
+
+import com.appcoins.wallet.core.network.backend.api.ImpressionApi
+import com.appcoins.wallet.core.utils.android_common.RxSchedulers
+import io.reactivex.Completable
+import javax.inject.Inject
+
+class ImpressionRepository @Inject constructor(
+  private val impressionApi: ImpressionApi,
+  private val rxSchedulers: RxSchedulers
+) {
+  
+  fun getImpression(): Completable {
+    return impressionApi.getImpression().subscribeOn(rxSchedulers.io)
```

Comment:
this should be "subscribeOn". the observeOn should be on main thread.

Reply:
In this line already is subscribeOn